lang/java/maven-plugin/src/main/java/org/apache/avro/mojo/AbstractAvroMojo.java

```diff
@@ -202,6 +203,13 @@ public abstract class AbstractAvroMojo extends AbstractMojo {
 
   @Override
   public void execute() throws MojoExecutionException {
+    if (Files.isSymbolicLink(sourceDirectory.toPath())) {
```

Comment:
The first line below is a null check, but this will throw if sourceDirectory is null.